### DISCUSSION TITLE
WARN com.diozero.sbc.UnknownBoardInfo.get

### DISCUSSION THREAD

**Discussion by Plamen22** [mattjlewis/diozero#191] [General]
Hi!

A week ago it was working without warning!

22:55:10.281 [main] WARN com.diozero.sbc.UnknownBoardInfo.get - Failed to resolve board info for hardware 'Raspberry Pi 4 Model B Rev 1.5' and revision 'c03115'. Local O/S: Linux-aarch64
22:55:10.521 [main] WARN com.diozero.internal.provider.builtin.DefaultDeviceFactory.start - GPIO pull-up / pull-down control is not available for this board

Maven project 
<dependency>
    <groupId>com.diozero</groupId>
    <artifactId>diozero-core</artifactId>
    <version>1.3.5</version>
</dependency>

**Comment by mattjlewis** [comment]
Strange - have tested with latest code that will soon form 1.4.0 and can't reproduce. Created a [test case](https://github.com/mattjlewis/diozero/blob/main/diozero-core/src/test/java/com/diozero/sbc/SystemInfoTest.java#L270) in SystemInfoTest to validate. Also 1.4.0 adds support for pull-up/down via GPIO chardev, not just via mmap control.

**Reply by Plamen22** [reply]
Is it possible that the problem is in Tinylog?

**Comment by mattjlewis** [comment]
Can you run this command please (this is from the 1.3.5 release):

`java -cp diozero-sampleapps-1.3.5.jar com.diozero.sampleapps.SystemInformation`

Output should be similar to this (your revision 'c03115' is for 4GB rev 1.5 4B):

<img width="523" alt="image" src="https://github.com/mattjlewis/diozero/assets/12659472/c707e061-4d92-45e5-94a4-f19ad8f2f161">

If you still get the errors please run with debug enabled:

`java -Dtinylog.level@com.diozero=debug -cp diozero-sampleapps-1.3.5.jar com.diozero.sampleapps.SystemInformation`

**Comment by Plamen22** [comment]
![image_2024-01-15_182303002](https://github.com/mattjlewis/diozero/assets/100608337/1b01c8d1-fdd5-4fde-811b-f08ea6bfe312)

I don't know if I'm doing it right ?

**Reply by mattjlewis** [reply]
Please extract the full distribution ZIP file and run from within that directory - it will need diozero-core-1.3.5.jar, diozero-sampleapps-1.3.5.jar, tinylog-api-2.5.0.jar, tinylog-impl-2.5.0.jar and jansi-2.4.0.jar to run

**Reply by Plamen22** [reply]
![test1](https://github.com/mattjlewis/diozero/assets/100608337/2834c51a-0d8f-4abd-baf5-4825fba3e0eb)

**Reply by mattjlewis** [reply]
Thanks for this, definitely something worth investigating here. Are you able to share the full text output from running the command please?

**Reply by Plamen22** [reply]
![test2](https://github.com/mattjlewis/diozero/assets/100608337/663f74e5-c030-41c0-9e44-95b03d7c812d)

**Comment by mattjlewis** [comment]
Thanks for your patience. I was able to reproduce this using 1.3.5 on Raspberry Pi OS Bookworm on Pi4. I can confirm that I do not experience this issue when running the soon to be released 1.4.0.

**Reply by Plamen22** [reply]
Should I go back to debian 11 bullseye, will that fix the problem?
Will we be able to take advantage of Maven Diozero 1.4.0 soon?

Sorry for the english 

Thank you for understanding

**Reply by mattjlewis** [reply]
I will definitely release 1.4.0 this week

**Reply by Plamen22** [reply]
Great :)

**Comment by mattjlewis** [comment]
Just released 1.4.0, please confirm if it resolves your problem.

**Reply by Plamen22** [reply]
I don't have time to test it at the moment, but I will write to you as soon as possible.
